docs/reference/repl.rst

```diff
@@ -0,0 +1,202 @@
+
+    >>> for i in range(3):
+    ...     _
+
+If you then enter an if statement, an additional level of indentation will be
+provided:
+
+    >>> for i in range(30):
+    ...     if i > 3:
+    ...         _
+
+Now enter ``break`` followed by RETURN and press BACKSPACE:
+
+    >>> for i in range(30):
+    ...     if i > 3:
+    ...         break
+    ...     _
+
+Finally type ``print(i)``, press RETURN, press BACKSPACE and press RETURN again:
+
+    >>> for i in range(30):
+    ...     if i > 3:
+    ...         break
+    ...     print(i)
+    ...
+    0
+    1
+    2
+    3
+    >>>
+
+Auto Completion
+---------------
+
+While typing a command at the REPL, if the line typed so far corresponds to
+the beginning of the name of something, then pressing TAB will show
+possible things that could be entered. For example type ``m`` and press TAB
+and it should expand to ``machine``. Enter a dot ``.`` and press TAB again. You
+should see something like:
+
+    >>> machine.
+    __name__        info            unique_id       reset
+    bootloader      freq            rng             idle
+    sleep           deepsleep       disable_irq     enable_irq
+    Pin
+
+The word will be expanded as much as possible until multiple possibilities exist.
+For example, type ``machine.Pin.AF3`` and press TAB and it will expand to
+``machine.Pin.AF3_TIM``. Pressing TAB a second time will show the possible
+expansions:
+
+    >>> machine.Pin.AF3_TIM
+    AF3_TIM10       AF3_TIM11       AF3_TIM8        AF3_TIM9
+    >>> machine.Pin.AF3_TIM
+
+Interrupting a running program
+------------------------------
+
+You can interupt a running program by pressing CTRL-C. This will raise a KeyboardInterrupt
+which will bring you back to the REPL, providing your program doesn't intercept the
+KeyboardInterrupt exception.
+
+For example:
+
+    >>> for i in range(1000000):
+    ...     print(i)
+    ...
+    0
+    1
+    2
+    3
+    ...
+    6466
+    6467
+    6468
+    Traceback (most recent call last):
+      File "<stdin>", line 2, in <module>
+    KeyboardInterrupt:
+    >>>
+
+Paste Mode
+----------
+
+If you want to paste some code into your terminal window, the auto-indent feature
+will mess things up. For example, if you had the following python code: ::
+
+   def foo():
+       print('This is a test to show paste mode')
+       print('Here is a second line')
+   foo()
+
+and you try to paste this into the normal REPL, then you will see something like
+this:
+
+    >>> def foo():
+    ...         print('This is a test to show paste mode')
+    ...             print('Here is a second line')
+    ...             foo()
+    ...
+    Traceback (most recent call last):
+      File "<stdin>", line 3
+    IndentationError: unexpected indent
+
+If you press CTRL-E, then you will enter paste mode, which essentially turns off
+the auto-indent feature, and changes the prompt from ``>>>`` to ``===``. For example:
+
+    >>>
+    paste mode; CTRL-C to cancel, CTRL-D to finish
```

Comment:
The repo still says CTRL:
https://github.com/micropython/micropython/blob/master/lib/utils/pyexec.c#L413
but I can change it....